contracts/providers/chainlink/ChainlinkPairAggregator.sol

```diff
@@ -38,15 +55,34 @@ contract ChainlinkPairAggregator is PairAggregatorInterface {
         return aggregator.latestRound();
     }
 
-    function getPreviousAnswer(uint256 roundsBack) external view returns (int256) {
+    function getPreviousTimestamp(uint256 roundsBack) external view returns (uint256) {
         uint256 latest = aggregator.latestRound();
-        require(roundsBack <= latest, "Not enough history");
-        return aggregator.getAnswer(latest - roundsBack);
+        require(roundsBack <= latest, "NOT_ENOUGH_HISTORY");
+        return aggregator.getTimestamp(latest - roundsBack);
     }
 
-    function getPreviousTimestamp(uint256 roundsBack) external view returns (uint256) {
+    /** Internal Functions */
+
+    function _getLatestAnswer() internal view returns (int256) {
```

Comment:
Why is this function factored out? It's only used once

Reply:
Correct. It's due to I had another impl before. 
Done